app/models/menu.rb

```diff
@@ -0,0 +1,52 @@
+# A Menu is a view into a hierarchy of pages.
+class Menu
+  class Item
+    def initialize(page, text = nil, include_children = false)
+      @page = page
+      @text = text
+      @include_children = include_children
+    end
+
+    def children
+      @children ||= if @include_children
+        @page.children.map { |child| Menu::Item.new child }
+      else
+        []
+      end
+    end
+
+    def has_children?
+      @include_children && @page.has_children?
+    end
+
+    def path
+      @page.filename
+    end
+
+    def public?
+      @page.public?
+    end
+
+    def text
+      @text || @page.title
+    end
+  end
+
+  def self.load(file, pages)
+    data = YAML.safe_load File.read(file), fallback: {}
+    items = (data["items"] || []).map { |i|
+      page = Page.find_by_slug(i["page"], pages)
+      if page
+        self::Item.new(page, i["text"], i["include_children"])
```

Comment:
[question] is `self` required there? I don't think it should be, but I could be wrong.